uniffi_bindgen/src/bindings/kotlin/gen_kotlin.rs

```diff
@@ -124,6 +125,14 @@ mod filters {
                 calculate_write_size(&"v", t)?,
                 write_kt(&"v", &"buf", t)?
             ),
+            Type::Map(t) => format!(
+                "lowerMap({}, {{ k, v -> {} + {} }}, {{ k, v, buf -> {}; {} }})",
+                nm,
+                calculate_write_size(&"k", &Type::String)?,
+                calculate_write_size(&"v", t)?,
+                write_kt(&"k", &"buf", &Type::String)?,
+                write_kt(&"v", &"buf", t)?
+            ),
```

Comment:
🤮

Reply:
lol, but also, I wonder if something similar to what I tried for python over in https://github.com/mozilla/uniffi-rs/pull/214 could be slightly less vomit-inducing here; thoughts? (for a future PR of course, not this one)

Reply:
What I'm concerned about is that we would generate a lot of extra code for each new record/sequence/etc?

Reply:
Yeah, I guess it's similar to the use of monomorphization in rust, where you rust generating lots of almost-duplicate code. (I don't think it would balloon too badly in practice, because you'd only get one method per type-you-actually-use-in-a-sequence, not one for every single type. But, worth watching out for).